app/controllers/characters_controller.rb

```diff
@@ -319,6 +319,6 @@ def build_template
   end
 
   def character_params
-    params.fetch(:character, {}).permit(:default_icon_id, :name, :template_name, :screenname, :setting, :template_id, :pb, :description, ungrouped_gallery_ids: [], gallery_group_ids: [], template_attributes: [:name, :id])
+    params.fetch(:character, {}).permit(:default_icon_id, :name, :template_name, :screenname, :template_id, :pb, :description, ungrouped_gallery_ids: [], setting_ids: [], gallery_group_ids: [], template_attributes: [:name, :id])
```

Comment:
This is getting long, would suggest putting the individual attributes on new lines (not a blocker or anything, minor)